### PR TITLE
Use Rust docstring comments

### PR DESCRIPTION
**Pull request by JasonGross**
Addresses part of #785

@mimoo , is this good?

Note that this conflicts with #786, and whichever is merged second will need to regenerate the rust files to include both changes.

### CODE REVIEW

File: fiat-rust/src/curve25519_64.rs [hunk 1]
```diff
@@ -17,20 +17,18 @@ pub type fiat_25519_u2 = u8;
...
+ //!
+ //! Input Bounds:
+ //!   arg1: [0x0 ~> 0x1]
+ //!   arg2: [0x0 ~> 0x7ffffffffffff]
+ //!   arg3: [0x0 ~> 0x7ffffffffffff]
+ //! Output Bounds:
+ //!   out1: [0x0 ~> 0x7ffffffffffff]
+ //!   out2: [0x0 ~> 0x1]
 #[inline]
 pub fn fiat_25519_addcarryx_u51(out1: &mut u64, out2: &mut fiat_25519_u1, arg1: fiat_25519_u1, arg2: u64, arg3: u64) -> () {
```
Comment (JasonGross) [inline fiat-rust/src/curve25519_64.rs:33]: @mimoo this errors with
```
error: expected outer doc comment
  --> src/curve25519_64.rs:20:2
   |
20 |  //! The function fiat_25519_addcarryx_u51 is an addition with carry.
   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: inner doc comments like this (starting with `//!` or `/*!`) can only appear before items

error: aborting due to previous error

error: could not compile `fiat-crypto`.
```
How is it supposed to look?

Comment (mimoo) [reply fiat-rust/src/curve25519_64.rs:33]: I'm so sorry, I meant `///` not `//!` :D 

`//!` is supposed to be used at the start of the file to provide "module documentation"

Comment (mimoo) [reply fiat-rust/src/curve25519_64.rs:33]: See this for an example: https://github.com/dalek-cryptography/curve25519-dalek/blob/master/src/scalar.rs

and the corresponding generated doc: https://doc.dalek.rs/curve25519_dalek/scalar/struct.Scalar.html

Comment (JasonGross) [reply fiat-rust/src/curve25519_64.rs]: Thanks!

**Pull request merged**